test/parallel/test-buffer-readintbe.js

```diff
@@ -0,0 +1,24 @@
+'use strict';
+require('../common');
+const assert = require('assert');
+
+// testing basic functionality of readIntBE()
+
+const buf = Buffer.from([42, 84, 167, 127]);
+const result = buf.readIntBE(1);
+
+assert.strictEqual(result, 42);
+
+assert.throws(
+  () => {
+    buf.readIntBE(5);
+  },
+  /^RangeError: Index out of range$/
+);
+
+assert.doesNotThrow(
+  () => {
+    buf.readIntBE(5, 0, true);
```

Comment:
Or @Trott, starting harder at the docs and re-reading your comment again, are you saying that noAssert applies explicitly only to the check against the offset argument, and not to the (apparently completely missing, eeks) check that should be present for the byteLength argument?